[ADD] Timesheets: configuration and basics (#16459)
odoo/documentation: content/applications/services/timesheets.rst

:show-content:

==========
Timesheets
==========

Odoo Timesheets lets you log, view, and validate working time across projects and employees.
Timesheets enable accurate time tracking and productivity monitoring, and in the case of services
invoiced on a time-and-material basis, they are essential for invoicing clients.

In Odoo, Timesheets is a standalone app, but they are also integrated into Project tasks and Helpdesk
tickets.

Configuring timesheets

> ======================

Reviewer: I think this heading isn't totally accurate since we're not configuring timesheets per say, we're just carrying out the configuration necessary for using and loggine timesheets, if this makes sense?

I'd rename this *Configuration*
and then I think it would make sense to have two sub-sections here: *Access rights* and *Project*

.. note::
   Odoo users must be registered as
   :doc:`employees </applications/hr/employees/new_employee>` to log
   timesheets.

To define what a user can view and edit when using the Timesheets application, the user's access

> rights must be set for the employee:

Reviewer: ```suggestion
To define what a user can view and edit in the Timesheets application, configure their access
rights.
```
- From what i tested, users that are not employee can still be administrator or have the user:all timesheets access rights so I think we can remove the employee mention in this sentence
- I think we can use the imperative here instead of the passive voice
- Period instead of colon to fit better with the next (changed) sentence

Open the :guilabel:`Settings` app and go to :menuselection:`Users & Companies --> Users`. Select the
specific user and under the :guilabel:`Services` section, set the :guilabel:`Timesheets` field to

> one of the following:

Reviewer: I'd add this directly after the previous sentence:

```suggestion
To do so, open the :guilabel:`Settings` app, go to :menuselection:`Users & Companies --> Users`, and select the relevant user.  Then, under the :guilabel:`Services` section, set the :guilabel:`Timesheets` field to one of the following:
```

- :guilabel:`No` (blank): The user cannot see the Timesheets app and cannot log time.
- :guilabel:`User: own timesheets only`: The user can record their own time, but cannot view anyone
  else’s entries.
- :guilabel:`User: all timesheets`: The user can record their own time and can view everyone else's
  entries.
- :guilabel:`Administrator`: The user can log their own time and view, edit, and validate their
  entire team's timesheets.

.. important::
   A user can only log time if the project itself allows it. To enable Timesheets in a project:

   - Go to the :guilabel:`Project` app, click the :icon:`fa-ellipsis-v`
     (:guilabel:`vertical ellipsis`) icon on a project card and select :guilabel:`Settings`.

>    - Click the :guilabel:`Settings` tab and ensure the :guilabel:`Timesheets` checkbox is enabled.

Reviewer: I'd add this under a heading instead of an important note (like I mentionned at the beginning of this section) and rephrase as follows:

```suggestion
   Employees can only log time if the project itself allows it. To enable timesheets for a project, follow these steps:
   
   #. Go to the :guilabel:`Project` app.
   #. Click the :icon:`fa-ellipsis-v` (:guilabel:`vertical ellipsis`) icon on the relevant project's card   and select :guilabel:`Settings`. 
   #. Go to the :guilabel:`Settings` tab and scroll down to the :guilabel:`Time management` section.
   #. Enable :guilabel:`Timesheets`.
```

Viewing timesheets
==================

By default, the Timesheets app opens to the :guilabel:`My Timesheets` screen in :guilabel:`Grid`

> view, displaying the time spent on projects, per day, in each cell.

Reviewer: I think I would also mention the Time Spent colum somewhere in this section maybe?

To change the displayed period, click on :guilabel:`Week`, then select :guilabel:`Day` or
:guilabel:`Month` in the dropdown menu. Use the left and right arrow buttons next to it to view
previous or upcoming timesheets.

Click :guilabel:`Today` to return to the current day while keeping you in the selected view.

.. image:: timesheets/timesheets.png
   :alt: Grid view in Timesheets.

.. tip::
   Timesheets displayed in italics are suggested entries: projects for which the user has not yet
   logged time for the period, and based on task assignments, deadlines, and historical timesheet
   data.

Hover your mouse over a cell in the grid view and click the magnifying glass icon to view and edit
details for that entry, including the :guilabel:`Date`, :guilabel:`Project`, :guilabel:`Task`,

> :guilabel:`Description`, :guilabel:`Sales Order Item`, and :guilabel:`Time Spent`.

Reviewer: ```suggestion
:guilabel:`Description`, :guilabel:`Sales Order Item` (if the project is billable --> add a ref here to the relevant doc), and :guilabel:`Time Spent`.
```

Author: I added _(if the project is billable)_, but no ref because there isn't a doc (yet) dedicated comprehensively on billable projects.

.. note::
   More views are available to display your timesheets. Click the buttons at the top right of the
   page to switch between :guilabel:`Grid`, :guilabel:`List`, :guilabel:`Calendar`,
   :guilabel:`Kanban`, :guilabel:`Pivot`, and :guilabel:`Graph` views.

Daily totals
------------

At the bottom of each column, the logged time is color-coded based on the user’s
:doc:`daily goal </applications/hr/employees/new_employee>`, calculated according to their working

> schedule:

Author: Much more clear indeed, thank you :)

- **Green**: Logged time matches the expected amount exactly.
- **Brown**: Logged time exceeds the expected amount.
- **Red**: Logged time is less than the expected amount.

.. tip::
   Hover over the purple section below the bottom line to view the specific daily overtime and
   undertime.

Period totals
-------------

The value in the color-coded cell at the bottom right of the chart represents the total time logged
for the period:

- **Green**: Total time matches the expected amount for the period exactly.
- **Orange**: Total time exceeds the expected amount for the period.
- **Red**: Total time is less than the expected amount for the period.

Logging timesheets
==================

Odoo offers multiple ways to log timesheets.

Using the timer
---------------

> The built-in timer allows you to record your timesheets in real time:

Reviewer: ```suggestion
To record the time spent on a project in real time, follow these steps: 
```

#. Go to the :guilabel:`Timesheets` app.
#. Click :guilabel:`Start` to launch the timer.
#. By default, the project most frequently used in the last five timesheet entries is automatically
   selected. To log time for another project, type or select it in the :guilabel:`Project` field.
   Optionally, :guilabel:`Describe your activity` and select a :guilabel:`Task` in the corresponding
   fields.

> #. Click :guilabel:`Stop` to end the timer once you’ve finished the work. The time spent is

Reviewer: ```suggestion
#. Click :guilabel:`Stop` to stop the timer. The time spent is
```

:show-content:

==========
Timesheets
==========

Odoo Timesheets lets you log, view, and validate working time across projects and employees.
Timesheets enable accurate time tracking and productivity monitoring, and in the case of

> :doc:`services invoiced on a time-and-material basis </applications/services/timesheets/billing_rates>`,

Reviewer: Here you can use a [relative path](https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#relative-links):

```suggestion
:doc:`services invoiced on a time-and-material basis <timesheets/billing_rates>`,
```
But actually, I'm not sure I understand the link between this topic and the linked doc. Is there a reason why you chose this particular doc instead of the one [I suggested in my previous comment](https://github.com/odoo/documentation/pull/16459#discussion_r2826088957)?

they are essential for invoicing clients.

.. note::
   Timesheets are also integrated into
   :doc:`Project tasks </applications/services/project/project_management/project_dashboard>` and
   :doc:`Helpdesk tickets </applications/services/helpdesk/advanced/track_and_bill>`.

Configuration
=============

Access rights
-------------

.. note::
   Odoo users must be registered as
   :doc:`employees </applications/hr/employees/new_employee>` to log timesheets.

To define what a user can view and edit in the Timesheets application, configure their access
rights. To do so, open the Settings app, go to :menuselection:`Settings --> Users & Companies -->
Users`, and select the relevant user. Then, under the :guilabel:`Services` section, set the
:guilabel:`Timesheets` field to one of the following:

- :guilabel:`No` (blank): The user cannot see the Timesheets app and cannot log time.
- :guilabel:`User: own timesheets only`: The user can log their own time, but cannot view anyone

>   else’s entries.

Author: Done! Removed 5 more of them

Daily totals
------------

At the bottom of each column, the logged time is color-coded based on the
:ref:`Working Hours defined for the employee <employees/schedule>`:

- **Green**: Logged time matches the expected amount exactly.
- **Brown**: Logged time exceeds the expected amount.
- **Red**: Logged time is less than the expected amount.

.. tip::
   Hover over the purple section below the bottom line to view the overtime/undertime recorded for
   that specific day.

Period totals
-------------

The value in the color-coded cell at the bottom of the :guilabel:`Time Spent` column represents the
total time logged for the period:

- **Green**: Total time matches the expected amount for the period exactly.
- **Orange**: Total time exceeds the expected amount for the period.
- **Red**: Total time is less than the expected amount for the period.

Logging timesheets
==================

Odoo offers multiple ways to log timesheets:

- :ref:`Using the timer <contributing/rst/timer>`
- :ref:`Manually <contributing/rst/log-manually>`
- :ref:`From Project and Helpdesk <contributing/rst/project-helpdesk>`

 .. _contributing/rst/timer:

> Using the timer

Reviewer: Great job on grouping everything timer-related under this section, thank you!
